packages/datadog-instrumentations/test/http.spec.js

```diff
@@ -0,0 +1,185 @@
+'use strict'
+
+const { assert } = require('chai')
+const dc = require('dc-polyfill')
+
+const agent = require('../../dd-trace/test/plugins/agent')
+describe('client', () => {
+  let url, http, startChannelCb, endChannelCb, asyncStartChannelCb, errorChannelCb
+
+  const startChannel = dc.channel('apm:http:client:request:start')
+  const endChannel = dc.channel('apm:http:client:request:end')
+  const asyncStartChannel = dc.channel('apm:http:client:request:asyncStart')
+  const errorChannel = dc.channel('apm:http:client:request:error')
+
+  before(async () => {
+    await agent.load('http')
+  })
+
+  after(() => {
+    return agent.close()
+  })
+
+  beforeEach(() => {
+    startChannelCb = sinon.stub()
+    endChannelCb = sinon.stub()
+    asyncStartChannelCb = sinon.stub()
+    errorChannelCb = sinon.stub()
+
+    startChannel.subscribe(startChannelCb)
+    endChannel.subscribe(endChannelCb)
+    asyncStartChannel.subscribe(asyncStartChannelCb)
+    errorChannel.subscribe(errorChannelCb)
+  })
+
+  afterEach(() => {
+    startChannel.unsubscribe(startChannelCb)
+    endChannel.unsubscribe(endChannelCb)
+    asyncStartChannel.unsubscribe(asyncStartChannelCb)
+    errorChannel.unsubscribe(errorChannelCb)
+  })
+
+  /*
+   * Necessary because the tracer makes extra requests to the agent
+   * and the same stub could be called multiple times
+   */
+  function getContextFromStubByUrl (url, stub) {
+    for (let i = 0; i < stub.args.length; i++) {
+      const arg = stub.args[i][0]
+      if (arg.args?.originalUrl === url) {
+        return arg
+      }
+    }
+
+    return null
+  }
```

Comment:
```suggestion
  function getContextFromStubByUrl (url, stub) {
    return stub.args.map(arg => arg[0])
        .find(arg => arg.args?.originalUrl === url)
  }
```

Reply:
Is a test, so the performance regression that you are suggesting is not a problem (and the list is going to be short), but do you think that it is easier to understand than the `for` that I made?

Reply:
the `for` is more "noisy", another option could be a for-of
```suggestion
  function getContextFromStubByUrl (url, stub) {
    for (const args of stub.args) {
      const arg = args[0]
      if (arg.args?.originalUrl === url) {
        return arg
      }
    }

    return null
  }
```

Reply:
Applied the second option